spark/src/test/scala/org/apache/comet/CometCastSuite.scala

```diff
@@ -807,11 +833,22 @@ class CometCastSuite extends CometTestBase with AdaptiveSparkPlanHelper {
             } else {
               // Spark 3.2 and 3.3 have a different error message format so we can't do a direct
               // comparison between Spark and Comet.
+              // In the case of CAST_INVALID_INPUT
               // Spark message is in format `invalid input syntax for type TYPE: VALUE`
               // Comet message is in format `The value 'VALUE' of the type FROM_TYPE cannot be cast to TO_TYPE`
               // We just check that the comet message contains the same invalid value as the Spark message
-              val sparkInvalidValue = sparkMessage.substring(sparkMessage.indexOf(':') + 2)
-              assert(cometMessage.contains(sparkInvalidValue))
+              // In the case of CAST_OVERFLOW
+              // Spark message is in format `Casting VALUE to TO_TYPE causes overflow`
+              // Comet message is in format `The value 'VALUE' of the type FROM_TYPE cannot be cast to TO_TYPE
+              // due to an overflow`
+              // We check if the comet message contains 'overflow'.
+              val sparkInvalidValue = if (sparkMessage.indexOf(':') == -1) {
+                EMPTY_STRING
+              } else {
+                sparkMessage.substring(sparkMessage.indexOf(':') + 2)
+              }
+              assert(
+                cometMessage.contains(sparkInvalidValue) || cometMessage.contains("overflow"))
```

Comment:
If `sparkInvalidValue` is `EMPTY_STRING`, won't `cometMessage.contains(sparkInvalidValue)` always be true?

Reply:
you are right, my bad 😅 . so incase sparkMessage doesnt have ':' should I assert on just commetMessage.contains("overflow")
something like this,

```
if sparkMessage.indexOf(':') == -1 then assert(commetMessage.contains("overflow"))
else assert(commetMessage.contains(sparkInvalidValue) || commetMessage.contains("overflow"))
```

Reply:
Yes, something like that. I haven't reviewed the overflow messages to see if they contain `:` though (in any of the spark versions 3.2, 3.3, and 3.4)

Reply:
doesnt look like overflow error message has ':' in it.

**3.4** - [CAST_OVERFLOW] The value 9223372036854775807L of the type "BIGINT" cannot be cast to "INT" due to an overflow. Use `try_cast` to tolerate overflow and return NULL instead. If necessary set "spark.sql.ansi.enabled" to "false" to bypass this error
**3.3** - The value 9223372036854775807L of the type "BIGINT" cannot be cast to "INT" due to an overflow. Use `try_cast` to tolerate overflow and return NULL instead. If necessary set "spark.sql.ansi.enabled" to "false" to bypass this error
**3.2** - Casting 9223372036854775807 to int causes overflow